open-craft/oauth-example
# [STAR-2807] feat: Oauth demo.

## Description

This PR has the demo code for the blog post on [using the LMS platform as an authentication provider](https://opencraft.com/blog/use-the-open-edx-lms-for-authentication/). That link won't work until the post is actually published, so check the link in the ticket for backend access to it.

https://tasks.opencraft.com/browse/STAR-2807

## Testing instructions

1. Install the requirements
2. follow the directions in the blogpost to set up on the LMS side
3. Update the settings.py file with your client_id and secret
4. `./manage.py runserver`
5. Visit the URL given in your terminal (should be http://127.0.0.1:8000/)
6. Authorize the app
7. See the username in a JSON object rendered!
